### DISCUSSION TITLE
Transit Networks & the choice of Speed Test Locations

### DISCUSSION THREAD

**Discussion by hohl** [masonr/yet-another-bench-script#49] [Ideas]
The current selection of `iperf`-servers has quite a bias to a small number of global transit networks. Especially Arelion (fka Telia) and Lumen, which is in part due to the strong presence of Clouvider (and an overweight of European providers in general).

| Network Provider | IP Transit Networks in Use |
| ------------- | ------------- |
| Clouvider UK | [Arelion, GTT & Lumen](https://bgp.tools/prefix/5.180.211.0/24) |
| Scaleway | [Arelion, Cogent, Orange & Lumen](https://bgp.tools/prefix/62.210.0.0/16) |
| NovoServe | [Arelion, Cogent, NTT, Liberty Global & Lumen](https://bgp.tools/prefix/185.80.232.0/22) |
| Uztelecom | [RETN & TATA](https://bgp.tools/prefix/195.69.189.0/24) |
| Clouvider NY | [Arelion & GTT](https://bgp.tools/prefix/94.154.159.0/24) |
| Clouvider TX | [Arelion, GTT & Lumen](https://bgp.tools/prefix/2.56.188.0/22) |
| Clouvider LA | [Arelion, GTT, Lumen & TI Sparkle](https://bgp.tools/prefix/77.247.126.0/24) |

I know the available choices of public `iperf` nodes is very limited—especially outside of Europe—, but I thought it might still be interesting to point this out. This could be a point to consider when adding/removing locations, so that we get some more diversity in the transit networks tested against.

Btw. I like the recent addition of Uztelecom, it gives at least some variation!

**Comment by masonr** [comment]
I agree with your assessment here and this is definitely a valid concern. Some variation is probably a good call to avoid any bias on transits.

I've been following this project [R0GGER/public-iperf3-servers](/R0GGER/public-iperf3-servers) pretty closely lately and like what I see so I'm thinking about making the iperf locations that YABS selects somewhat dynamic or random using locations pulled from that project. 

Just an idea I've been sitting on for a bit, but I'm open to new ideas too.

**Comment by febryanvaldo** [comment]
Hello, i'm pretty new to this benchmark script, as @hohl pointed out, i also want to suggest if you can add some Asia locations (Singapore, Jakarta, Australia or Hongkong if possible). Thanks.

**Reply by masonr** [reply]
Finding reliable iperf3 APAC locations has been really difficult the past couple years. Seems every time I find a nice one to add, it gets taken down or overloaded soon after. I'll be testing out some of the entries from here - https://github.com/R0GGER/public-iperf3-servers#asia - to see if there's any reliable ones that could be thrown in.

**Reply by febryanvaldo** [reply]
Well, looking forward for that. Really appreciate it. Thanks.
